mysqld_exporter.go

```diff
@@ -124,40 +126,85 @@ func (e *Exporter) scrape(scrapes chan<- []string) {
 	// fetch slave status
 	rows, err = db.Query("SHOW SLAVE STATUS")
 	if err != nil {
-		log.Println("error running show slave query on database: ", err)
+		log.Println("error running show slave query on database:", err)
 		e.error.Set(1)
 		e.duration.Set(float64(time.Now().UnixNano()-now) / 1000000000)
 		return
 	}
 	defer rows.Close()
 
+	var slaveCols []string
+
+	slaveCols, err = rows.Columns()
+	if err != nil {
+		log.Println("error retrieving column list:", err)
+		e.error.Set(1)
+		e.duration.Set(float64(time.Now().UnixNano()-now) / 1000000000)
+		return
+	}
+
+	var slaveData = make([]sql.RawBytes, len(slaveCols))
+
+	// As the number of columns varies with mysqld versions,
+	// and sql.Scan requires []interface{}, we need to create a
+	// slice of pointers to the elements of slaveData.
+
+	scanArgs := make([]interface{}, len(slaveCols))
+	for i := range slaveData {
+		scanArgs[i] = &slaveData[i]
+	}
+
 	for rows.Next() {
-		// get RawBytes from data
-		err = rows.Scan(&key, &val)
+
+		err = rows.Scan(scanArgs...)
 		if err != nil {
-			log.Printf("error getting result set: ", err)
+			log.Println("error retrieving result set:", err)
+			e.error.Set(1)
+			e.duration.Set(float64(time.Now().UnixNano()-now) / 1000000000)
 			return
 		}
 
+	}
+
+	threadRexp := regexp.MustCompile("^Slave_(.*)_Running$")
```

Comment:
You could combine all bar the log_file into one regexp, which'll reduce the copy&paste